### PR TITLE
Add 'metadata_plus_exceptions' function to return SensorML, plus ServiceExceptions

### PR DESCRIPTION
**Pull request by mwengren**
Some SOS services do not always successfully return SensorML for all procedures, breaking 'metadata' function.

### CODE REVIEW

**Comment by daf** [comment]
Looks like the build failures here exist outside of the PR too.  Can you perhaps add a test?

File: pyoos/collectors/ioos/swe_sos.py [hunk 1]
```diff
@@ -34,6 +35,35 @@ def metadata(self, output_format=None, feature_name_callback=None, **kwargs):
...
+            output_format = 'text/xml; subtype="sensorML/1.0.1/profiles/ioos_sos/1.0"'
+
+        responses = []
+        response_failures = {}
+        if self.features is not None:
+            for feature in self.features:
+                ds_kwargs = kwargs.copy()
+                ds_kwargs.update({'outputFormat': output_format,
+                                  'procedure'   : callback(feature)})
+                print("swe_sos.py metadata: {feature}".format(feature=feature))
```
Comment (daf) [inline pyoos/collectors/ioos/swe_sos.py]: this is a debug print?

Comment (mwengren) [inline pyoos/collectors/ioos/swe_sos.py]: Oops, meant to remove that.  Will fix.

**Comment by mwengren** [comment]
About the tests, I looked at what's already available for this class, which is nothing, except for the superclass, which is [here](https://github.com/ioos/pyoos/blob/master/tests/collectors/test_collector.py) I believe?.   There are pretty good tests for the subclasses of IoosSweSos though: [Coops](https://github.com/ioos/pyoos/blob/master/tests/collectors/test_coops_sos.py) and [Ndbc](https://github.com/ioos/pyoos/blob/master/tests/collectors/test_ndbc_sos.py), so I skipped adding anything to the superclass tests at first.  Should there be a new test class created for IoosSweSos?

**Comment by mwengren** [comment]
@daf I added a simple test class for IoosSweSos that hits the NANOOS SOS service and does some simple checks.  A lot of the other tests are still failing, and I don't think I can fix those at this point.  

Can we merge this with these tests added?

**Comment by daf** [comment]
@mwengren this is good, and will do a good job detecting drift of live services vs the pyoos codebase (aka, exactly the reason we have build failures currently, although those are data based and not format/processing based).  A level-up on this would be to capture the output of this service to embed in the test codebase as a string and plug it through the parsing pipeline to make sure the parsing is working as intended.  I'll merge this though.

**Pull request merged**